Blog/views.py

```diff
@@ -49,14 +61,22 @@ def get(self, request, id):
 class EditPage(View):
     ctx = {'id': 'edit_post', 'button': 'Edit and Save'}
 
-    def get(self, request, id, secret_key):
-        blog = get_object_or_404(Blog, id=id, secret_key=secret_key)
-        form = BlogForm(instance=blog)
+    def get(self, request, id, secret_key=None):
+        if secret_key is None:
```

Comment:
Check that if `/1/edit/secret` is given the post is anonymous. If `/1/edit` is given, check that the post author is same as request author. And put the common code in dispatch method instead of duplicating in get and post.